backend/tests/Functional/FishbowlWorkflowFunctionalTest.php

```diff
@@ -0,0 +1,265 @@
+        $gql = <<<GQL
+          mutation RunFishbowl(\$slug: String!) {
+            runFishbowl(input: {slug: \$slug}) {
+              fishbowl {
+                currentStatus
+              }
+            }
+          }
+        GQL;
+
+        $graphqlResponse = $this->callGraphqlWithToken($gql, $this->adminToken);
+        $this->assertSame(Fishbowl::STATUS_RUNNING, $graphqlResponse['runFishbowl']['fishbowl']['currentStatus']);
+    }
+
+    /** @test */
+    public function itFinishesFishbowlWithWrongUser(): void
+    {
+        self::bootKernel();
+
+        parent::setUp();
+
+        $this->createFishbowlWithStatus(Fishbowl::STATUS_RUNNING, $this->adminUser);
+
+        $gql = <<<GQL
+          mutation FinishFishbowl(\$slug: String!) {
+            finishFishbowl(input: {slug: \$slug}) {
+              fishbowl {
+                currentStatus
+              }
+            }
+          }
+        GQL;
+
+        $graphqlResponse = $this->callGraphqlWithToken($gql, $this->token);
+        $this->assertNull($graphqlResponse['finishFishbowl']['fishbowl']);
+    }
+
+    /** @test */
+    public function itFinishesFishbowl(): void
+    {
+        self::bootKernel();
+
+        parent::setUp();
+
+        $this->createFishbowlWithStatus(Fishbowl::STATUS_RUNNING, $this->adminUser);
+
+        $gql = <<<GQL
+          mutation FinishFishbowl(\$slug: String!) {
+            finishFishbowl(input: {slug: \$slug}) {
+              fishbowl {
+                currentStatus
+              }
+            }
+          }
+        GQL;
+
+        $graphqlResponse = $this->callGraphqlWithToken($gql, $this->adminToken);
+        $this->assertSame(Fishbowl::STATUS_FINISHED, $graphqlResponse['finishFishbowl']['fishbowl']['currentStatus']);
+    }
+
+    /** @test */
+    public function itCantRunFishbowlWithoutIntroduction(): void
+    {
+        self::bootKernel();
+
+        parent::setUp();
+
+        $this->createFishbowlWithStatus(Fishbowl::STATUS_NOT_STARTED, $this->adminUser);
+
+        $gql = <<<GQL
+          mutation RunFishbowl(\$slug: String!) {
+            runFishbowl(input: {slug: \$slug}) {
+              fishbowl {
+                currentStatus
+              }
+            }
+          }
+        GQL;
+
+        $graphqlResponse = $this->callGraphqlWithToken($gql, $this->adminToken);
+
+        $this->assertNull($graphqlResponse['runFishbowl']['fishbowl']);
+    }
+
+    private function logIn(User $user): string
+    {
+        $response = static::createClient()->request('POST', '/login', ['json' => [
+            'email' => $user->getEmail(),
+            'password' => 'admin',
+        ]]);
+
+        $this->assertResponseIsSuccessful();
+
+        $logInResponse = $response->toArray();
+
+        $this->assertArrayHasKey('token', $logInResponse);
+        $this->assertNotEmpty($logInResponse['token']);
+
+        return $logInResponse['token'];
+    }
+
+    /*** @return array<string, array<string, null|array<string, string>>> */
```

Comment:
@jordisala1991 Any help with the phpstan error
`return  
         type has no value type specified in iterable type array.               `